worldedit-core/src/main/java/com/sk89q/worldedit/EditSession.java

```diff
@@ -2419,31 +2500,47 @@ public int deformRegion(final Region region, final Vector3 zero, final Vector3 u
      * The Expression class is subject to change. Expressions should be provided via the string overload.
      * </p>
      */
+    @Deprecated
     public int deformRegion(final Region region, final Vector3 zero, final Vector3 unit, final Expression expression,
                             final int timeout) throws ExpressionException, MaxChangedBlocksException {
+        final Transform transform = new SimpleTransform(zero, unit);
+        return deformRegion(region, transform, expression, timeout, world, transform);
+    }
+
+    /**
+     * Internal version of {@link EditSession#deformRegion(Region, Vector3, Vector3, String, int)}.
+     *
+     * <p>
+     * The Expression class is subject to change. Expressions should be provided via the string overload.
+     * </p>
+     */
+    public int deformRegion(final Region region, final Transform outputTransform, final Expression expression,
+                            final int timeout, InputExtent inputExtent, final Transform inputTransform) throws ExpressionException, MaxChangedBlocksException {
         final Variable x = expression.getSlots().getVariable("x")
             .orElseThrow(IllegalStateException::new);
         final Variable y = expression.getSlots().getVariable("y")
             .orElseThrow(IllegalStateException::new);
         final Variable z = expression.getSlots().getVariable("z")
             .orElseThrow(IllegalStateException::new);
 
-        final WorldEditExpressionEnvironment environment = new WorldEditExpressionEnvironment(this, unit, zero);
+        final WorldEditExpressionEnvironment environment = new WorldEditExpressionEnvironment(this, outputTransform);
         expression.setEnvironment(environment);
 
         final DoubleArrayList<BlockVector3, BaseBlock> queue = new DoubleArrayList<>(false);
 
+        final Transform outputTransformInverse = outputTransform.inverse();
         for (BlockVector3 position : region) {
-            // offset, scale
-            final Vector3 scaled = position.toVector3().subtract(zero).divide(unit);
-
             // transform
+            final Vector3 scaled = outputTransformInverse.apply(position.toVector3());
```

Comment:
is `scaled` the correct name for this field now, if it's also translated? would `transformedPosition` be a more descriptive name?

Reply:
It's called `inputPosition` now and after the expression is evaluated, I introduced an intermediate variable called `outputPosition` to match it.
Resolving conversation